qiling/arch/arch.py

```diff
@@ -37,10 +51,10 @@ def stack_push(self, value: int) -> int:
         Returns: the top of stack after pushing the value
         """
 
-        self.ql.reg.arch_sp -= self.ql.pointersize
-        self.ql.mem.write(self.ql.reg.arch_sp, self.ql.pack(value))
+        self.regs.arch_sp -= self.pointersize
```

Comment:
Is it possible that some arch has a stack implementation that grows up? Not urgent for now, just some random thoughts, maybe we can leave a TODO here?

Reply:
I am really struggling with these stack accessors, as they clearly not belong to `QlArch`, but I don't know exactly where to put them. Anyway, I agree that this behavior should be overload-able.

Reply:
I agree. I suppose it should go to `ql.mem`?

Reply:
This is one of the options.. but it makes `mem` depend on `arch`, which does not sit well with me.
Another option would be `QlOs` that may maintain a stack object providing these methods. Although stack does not belong to OS but to a process, we don't have an object for process - and the closest one we have is `QlOS`.
The last reasonable option would be `Qiling`, which is not a good idea for several reasons..

I did say it's a struggle (: